pilot/pkg/networking/core/v1alpha3/httproute.go

```diff
@@ -237,10 +245,19 @@ func selectVirtualServices(virtualServices []config.Config, servicesByName map[h
 				break
 			}
 
-			for svcHost := range servicesByName {
-				if host.Name(h).Matches(svcHost) {
-					match = true
-					break
+			if host.Name(h).IsWildCarded() {
+				for svcHost := range servicesByName {
+					if host.Name(h).Matches(svcHost) {
+						match = true
+						break
+					}
+				}
+			} else {
+				for _, svcHost := range wildCardedSvcHosts {
+					if host.Name(h).Matches(svcHost) {
+						match = true
+						break
```

Comment:
Not understand why vs wildecar host match all service host, while fqdn matches only wildcard svc host?

Don't you miss wildcard match wildcard?

Reply:
Cases:
1. `wildcard vs host` and `wildcard svc host`
2. `wildcard vs host` and `fqdn svc host`
3. `fqdn vs host` and `wildcard svc host`
4. `fqdn vs host` and `fqdn svc host`

I think `vs wildecar host match all service host` covers case 1 and case 2.

`fqdn matches only wildcard svc host` covers case 3.

And map search covers case 4

```go
if servicesByName[host.Name(h)] != nil {
    match = true
    break
}
```

The aim of this PR is to remove case 4 in the loop because map search already handle it and it's the most usage case.